### PR TITLE
Update & cleanup project structure

### PR DESCRIPTION
**Pull request by encounter**
This updates the project structure to match the latest dtk-template, along with some opinionated changes & cleanup.

Mainly:
- Compilers download URL no longer uses discord
- Added asm build steps (for modding and matching asm files)
- README updates & remove outdated info

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -416,19 +434,11 @@
             Object(Matching, "zlib/adler32.c"),
             Object(Matching, "zlib/crc32.c"),
             Object(Matching, "zlib/deflate.c"),
+            Object(Matching, "zlib/trees.c"),
+            Object(Matching, "zlib/zutil.c"),
             Object(Matching, "zlib/inflate.c"),
+            Object(NonMatching, "zlib/inftrees.c"),
```
Comment (DarkRTA) [inline configure.py:440]: building this with a different compiler was intentional

whats the rationale behind this

Comment (encounter) [reply configure.py:440]: I just traced it down and it looks like Wii 1.0 got messed up in the current compilers.zip, will fix

Comment (encounter) [reply configure.py]: Fixed by using 1.0a

**Review by DarkRTA — APPROVED** [review]
lgtm

**Pull request merged**